### PR TITLE
cmd/charmcmd: Allow multiple users in list.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

File: cmd/charm/charmcmd/list.go [hunk 1]
```diff
@@ -71,24 +73,28 @@ func (c *listCommand) Run(ctxt *cmd.Context) error {
...
+	users := strings.Split(c.users, ",")
+	err = validateNames(users)
 	if err != nil {
 		return errgo.Mask(err)
 	}
-
-	path := "/list?owner=" + c.user + "&sort=name,-series"
+	v := url.Values{}
+	v.Set("sort", "name,-series")
+	for _, user := range users {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/list.go]: v["owner"] = users

?

Or just:

    v := url.Values{
        "sort": "name,-series",
        "owner": users,
    }

?

File: cmd/charm/charmcmd/list.go [hunk 2]
```diff
@@ -71,24 +73,28 @@ func (c *listCommand) Run(ctxt *cmd.Context) error {
...
+		v.Add("owner", user)
+	}
+	path := "/list?" + v.Encode()
 	var resp params.ListResponse
 	err = client.Get(path, &resp)
 	if err != nil {
-		return errgo.Notef(err, "cannot list for user %s", path)
+		return errgo.Notef(err, "cannot list for user(s) %s", path)
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/list.go]: Why include the whole path here rather than just the users as the message implies?

File: dependencies.tsv [hunk 3]
```diff
@@ -51,18 +51,18 @@ golang.org/x/net	git	d25186b37f34ebdbbea8f488ef055638dfab272d	2018-03-06T06:01:5
...
+gopkg.in/juju/charmrepo.v3	git	1068bda266ca4f268822587b820bf1b4a5438745	2018-06-19T10:58:31Z
 gopkg.in/juju/charmrepo.v4	git	09b3570af7ab9c2da5af8bf45c12c0c3986540cb	2018-06-19T14:40:43Z
-gopkg.in/juju/charmstore.v5	git	9ea0c22208d71cf21702966903143fe5ad22a731	2018-06-26T14:38:13Z
+gopkg.in/juju/charmstore.v5	git	6dc8626f4616876c51e6bea87c8a49e0d5949384	2018-07-19T13:35:12Z
 gopkg.in/juju/environschema.v1	git	7359fc7857abe2b11b5b3e23811a9c64cb6b01e0	2015-11-04T11:58:10Z
 gopkg.in/juju/idmclient.v1	git	203d20774ce8aeaa6f662ee0b48f8cb5bc0816a1	2018-03-20T16:18:56Z
-gopkg.in/juju/jujusvg.v3	git	6f7342099e20c84f560bd9fc8afe96ff7486613e	2017-11-14T17:07:01Z
-gopkg.in/juju/names.v2	git	c43e8bdf2f4915a7019a2f8ad561af1498731a21	2018-05-16T01:04:14Z
+gopkg.in/juju/jujusvg.v3	git	b69768dfdf884defd0eaf7b7f3c4e04372922d05	2018-06-28T07:42:25Z
+gopkg.in/juju/names.v2	git	54f00845ae470a362430a966fe17f35f8784ac92	2017-11-13T11:20:47Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 4]
```diff
@@ -51,18 +51,18 @@ golang.org/x/net	git	d25186b37f34ebdbbea8f488ef055638dfab272d	2018-03-06T06:01:5
...
-gopkg.in/goose.v2	git	36df5d12fc6d0ef1c102e1c18a22ddf345b18821	2018-03-22T12:54:45Z
-gopkg.in/httprequest.v1	git	7d6c4413c5fe1b6d1fee366252d42bc8bae66420	2018-03-19T13:12:14Z
-gopkg.in/juju/charm.v6	git	0f7685c8057defe2282747a237e119d18fcf8e20	2018-06-28T06:52:20Z
+gopkg.in/goose.v2	git	4dc44b23a313a8a3e4051e7ee568a954a8cb3385	2018-02-08T12:05:40Z
+gopkg.in/httprequest.v1	git	3531529dedf047a744dd77f5262515aefff8cd48	2018-03-19T12:54:57Z
+gopkg.in/juju/charm.v6	git	4c7183c1a6386a8865fa6c7ddbd3de70d05f1f82	2018-06-28T05:32:07Z
 gopkg.in/juju/charm.v6-unstable	git	514bb811b021ebeb3e7ffcf1c267e9803968f59d	2017-07-28T19:41:00Z
 gopkg.in/juju/charmrepo.v2-unstable	git	e79aa298df89ea887c9bffec46063c24bfb730f7	2016-11-17T15:25:28Z
-gopkg.in/juju/charmrepo.v3	git	6180e92d72b33ae3b510c9a1dc5348403a96252a	2018-06-28T06:52:36Z
+gopkg.in/juju/charmrepo.v3	git	1068bda266ca4f268822587b820bf1b4a5438745	2018-06-19T10:58:31Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 5]
```diff
@@ -51,18 +51,18 @@ golang.org/x/net	git	d25186b37f34ebdbbea8f488ef055638dfab272d	2018-03-06T06:01:5
 golang.org/x/sys	git	37707fdb30a5b38865cfb95e5aab41707daec7fd	2018-02-02T13:58:01Z
 gopkg.in/check.v1	git	4f90aeace3a26ad7021961c297b22c42160c7b25	2016-01-05T16:49:36Z
 gopkg.in/errgo.v1	git	442357a80af5c6bf9b6d51ae791a39c3421004f3	2016-12-22T12:58:16Z
-gopkg.in/goose.v2	git	36df5d12fc6d0ef1c102e1c18a22ddf345b18821	2018-03-22T12:54:45Z
-gopkg.in/httprequest.v1	git	7d6c4413c5fe1b6d1fee366252d42bc8bae66420	2018-03-19T13:12:14Z
-gopkg.in/juju/charm.v6	git	0f7685c8057defe2282747a237e119d18fcf8e20	2018-06-28T06:52:20Z
+gopkg.in/goose.v2	git	4dc44b23a313a8a3e4051e7ee568a954a8cb3385	2018-02-08T12:05:40Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 6]
```diff
@@ -51,18 +51,18 @@ golang.org/x/net	git	d25186b37f34ebdbbea8f488ef055638dfab272d	2018-03-06T06:01:5
...
+gopkg.in/httprequest.v1	git	3531529dedf047a744dd77f5262515aefff8cd48	2018-03-19T12:54:57Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 7]
```diff
@@ -75,4 +75,4 @@ gopkg.in/natefinch/lumberjack.v2	git	df99d62fd42d8b3752c8a42c6723555372c02a03	20
 gopkg.in/retry.v1	git	2d7c7c65cc71d024968d9ff4385d5e7ad3a83fcc	2018-01-16T15:34:15Z
 gopkg.in/tomb.v1	git	dd632973f1e7218eb1089048e0798ec9ae7dceb8	2014-10-24T13:56:13Z
 gopkg.in/tomb.v2	git	d5d1b5820637886def9eef33e03a27a9f166942c	2016-12-08T15:16:19Z
-gopkg.in/yaml.v2	git	5420a8b6744d3b0345ab293f6fcba19c978f1183	2018-03-28T19:50:20Z
+gopkg.in/yaml.v2	git	7f97868eec74b32b0982dd158a51a446d1da7eb5	2018-02-23T19:12:37Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 8]
```diff
@@ -26,11 +26,11 @@ github.com/juju/names	git	8a0aa0963bbacdc790914892e9ff942e94d6f795	2016-03-30T15
 github.com/juju/persistent-cookiejar	git	d67418f14c93a698e37b52468958d5d4dcf8a7dd	2017-04-28T16:15:59Z
 github.com/juju/plans-client	git	e9ac8ae4f906488362ccde8c8ece344e817eecae	2017-12-05T16:53:23Z
 github.com/juju/retry	git	62c62032529169c7ec02fa48f93349604c345e1f	2015-10-29T02:48:21Z
-github.com/juju/schema	git	075de04f9b7d7580d60a1e12a0b3f50bb18e6998	2016-04-20T04:42:03Z
+github.com/juju/schema	git	e4f08199aa80d3194008c0bd2e14ef5edc0e6be6	2018-01-09T04:18:50Z
 github.com/juju/terms-client	git	68c3a74aeab3633d1eede0c98a1ef62d59cbf6be	2017-12-04T17:53:31Z
-github.com/juju/testing	git	72703b1e95eb8ce4737fd8a3d8496c6b0be280a6	2018-05-17T13:41:05Z
+github.com/juju/testing	git	44801989f0f7f280bd16b58e898ba9337807f147	2018-04-02T13:06:37Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 9]
```diff
@@ -26,11 +26,11 @@ github.com/juju/names	git	8a0aa0963bbacdc790914892e9ff942e94d6f795	2016-03-30T15
...
 github.com/juju/usso	git	5b79b358f4bb6735c1b00f6ad051c07c1a1a03e9	2016-04-18T12:10:39Z
-github.com/juju/utils	git	57d958857adc4e7fe33ea74366aeee0f4a3a8dde	2018-05-17T03:57:10Z
+github.com/juju/utils	git	d18e608d01400189bcda3e2669505cbd30e9dda9	2018-02-07T02:18:10Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

File: dependencies.tsv [hunk 10]
```diff
@@ -41,7 +41,7 @@ github.com/mattn/go-colorable	git	ed8eb9e318d7a84ce5915b495b7d35e0cfe7b5a8	2016-
 github.com/mattn/go-isatty	git	66b8e73f3f5cda9f96b69efd03dd3d7fc4a5cdb8	2016-08-06T12:27:52Z
 github.com/mattn/go-runewidth	git	d96d1bd051f2bd9e7e43d602782b37b93b1b5666	2015-11-18T07:21:59Z
 github.com/matttproud/golang_protobuf_extensions	git	c12348ce28de40eed0136aa2b644d0ee0650e56c	2016-04-24T11:30:07Z
-github.com/prometheus/client_golang	git	d49167c4b9f3c4451707560c5c71471ff5291aaa	2018-03-19T13:17:21Z
+github.com/prometheus/client_golang	git	575f371f7862609249a1be4c9145f429fe065e32	2016-11-24T15:57:32Z
```
Comment (rogpeppe) [inline dependencies.tsv]: This looks like it's gone back in time.

**Review by rogpeppe — CHANGES_REQUESTED** [review]
LGTM modulo the dependency issues and a couple of minor suggestions, thanks.

**Review by mhilton — COMMENTED** [review]
Code looks fine, but why not fix #11 while you're in the code.

**Comment by rogpeppe** [comment]
LGTM thanks!

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**